pyonepassword/_py_op_deprecation.py

```diff
@@ -34,7 +34,9 @@ def wrapped(*args, **kwargs):
 
         wrapped.__name__ = '__init__'
         wrapped.__doc__ = self._update_doc(init.__doc__)
-        wrapped.deprecated_original = init
+
+        # mypy doesn't like wrapped.deprecated_original = init
+        setattr(wrapped, "deprecated_original", init)
```

Comment:
I usually solve this with something like

```
import typing as t


class WithDeprecatedOriginal(t.Protocol):
    deprecated_original: t.Callable

t.cast(wrapped, WithDeprecatedOriginal).deprecated_original = init
```

if you also add `@t.runtime_checkable` decorator to the protocol class you can use it in an isinstance at other points in code to see if an object has the correct property and have type checking on accessing that property.

Likely overkill here though :p

Reply:
interesting. thanks for the tip. I'm still mulling over how I feel about typing.cast in python. My years in C development and security auditing have left me with come pretty burned-in feelings.

I know the implications are totally different here, but It still feels like a "I couldn't solve it some other way" tool.

Interesting about the `runtime_checkable`. even if overkill here, good to know. Thanks